keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/kep.yaml

```diff
@@ -0,0 +1,44 @@
+title: Add CDI devices to device plugin API
+kep-number: 4009
+authors:
+  - "@elezar"
+owning-sig: sig-node
+participating-sigs: []
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+creation-date: 2023-05-16
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+  - "/keps/sig-node/3063-dynamic-resource-allocation"
+  - "/keps/sig-node/3573-device-plugin"
+replaces: []
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha|beta|stable
```

Comment:
This should be `alpha`.

Reply:
Will update. One question I had was what stage this should be if the device plugin API is currently beta, or is it ok to have an alpha feature in a beta API?

Reply:
Even though this feature is making changes to device plugin API which is currently Beta, it would have its own lifecycle and graduate from alpha (where feature gate is disabled by default to preserve default behavior) to Beta (feature gate enabled by default when this functionality is well tested) to GA (no feature gate).
We had a similar case where we were contributing a new endpoint to a stable PodResource API and wanted to introduce a new endpoint.  I think we would have to do something similar here. Here is an example: https://github.com/kubernetes/enhancements/pull/2404. If you would like, you can double check this with [PRR reviewers](https://kubernetes.slack.com/archives/CPNHUMN74) to be sure.